src/pages/en/core-concepts/endpoints.md

```diff
@@ -0,0 +1,275 @@
+
+export async function get({ params }) {
+  const id = params.id;
+  const product = await getProduct(id);
+
+  if (!product) {
+    return new Response(null, {
+      status: 404,
+      statusText: 'Not found'
+    });
+  }
+
+  return new Response(JSON.stringify(product), {
+    status: 200,
+    headers: {
+      "Content-Type": "application/json"
+    }
+  });
+}
+```
+This will respond to any request that matches the dynamic route. For example, if we navigate to `/helmet.json`, `params.id` will be set to `helmet`. If `helmet` exists in the mock product database, the endpoint will use create a `Response` object to respond with JSON and return a successful [HTTP status code](https://developer.mozilla.org/en-US/docs/Web/API/Response/status). If not, it will use a `Response` object to respond with a `404`.
+
+
+### HTTP methods
+In addition to the `get` function, you can export a function with the name of any [HTTP method](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods). When a request comes in, Astro will check the method and call the corresponding function. 
+
+You can also export an `all` function to match any method that doesn't have a corresponding exported function. If there is a request with no matching method, it will redirect to your site's [404 page](http://localhost:3001/en/core-concepts/astro-pages/#custom-404-error-page).
+
+:::note
+Since `delete` is a reserved word in JavaScript, export a `del` function to match the delete method.
+:::
+
+```ts title="src/pages/methods.json.ts"
+export const get: APIRoute = ({ params, request }) => {
+  return {
+    body: JSON.stringify({
+      message: "This was a GET!"
+    })
+  }
+};
+
+export const post: APIRoute = ({ request }) => {
+  return {
+    body: JSON.stringify({
+      message: "This was a POST!"
+    })
+  }
+}
+
+export const del: APIRoute = ({ request }) => {
+  return {
+    body: JSON.stringify({
+      message: "This was a DELETE!"
+    })
+  }
+}
+
+export const all: APIRoute = ({ request }) => {
+  return {
+    body: JSON.stringify({
+      message: `This was a ${request.method}!`
+    })
+  }
+}
+```
+
+### `request` 
+In SSR mode, the `request` property returns a fully usable [`Request`](https://developer.mozilla.org/en-US/docs/Web/API/Request) object that refers to the current request. This allows you to accept data and check headers:
+
+```ts title="src/pages/test-post.json.ts"
+export const post: APIRoute = async ({ request }) => {
+  if (request.headers.get("Content-Type") === "application/json") {
+    const body = await request.json();
+    const name = body.name;
+    return new Response(JSON.stringify({
+      message: "Your name was: " + name
+    }), {
+      status: 200
+    })
+  }
+  return new Response(null, { status: 400 });
+}
+```
+
+### Redirects
+Since `Astro.redirect` is not available in API Routes you can use [`Response.redirect`](https://developer.mozilla.org/en-US/docs/Web/API/Response/redirect) to redirect:
```

Comment:
```suggestion
Since `Astro.redirect` is currently not yet available in API Routes you can use [`Response.redirect`](https://developer.mozilla.org/en-US/docs/Web/API/Response/redirect) to redirect:
```
Per Matthew's comment, we could imply that this may yet change, if you wanted.

Reply:
I think we can just document it as-is, and change it when needed